tests/nc-server/simple_server.h

```diff
@@ -0,0 +1,16 @@
+#ifndef SIMPLE_SERVER_H
+#define SIMPLE_SERVER_H
+
+#include <stdint.h>
+
+/**
+ * simple_server_init(port, nconn_max, shutdown_after, callback, caller_cookie):
+ * Run a server which accepts up to ${nconn_max} connections to port ${port}.
+ * After receiving a message, call ${callback} and pass it the
+ * ${caller_cookie}, along with the message.  Automatically shuts itself down
+ * after ${shutdown_after} connections have been dropped.
+ */
+int simple_server(const uint16_t, const size_t, const size_t,
```

Comment:
`port` should be a `in_port_t`.  But why not simply pass the entire address string in?